documentation/userguide/docs/metadata_forms.md

```diff
@@ -0,0 +1,104 @@
+# **Metadata Forms**
+
+**Introduction**
+
+Metadata forms allow users to add structured contextual information to various entities in the data.all platform. By creating and attaching metadata forms, user can standardize and enrich metadata in a customizable way.
+
+Metadata forms serve several key purposes:
+
+- Improve data discovery by enabling more consistent, complete, and meaningful metadata
+- Capture domain-specific or organizational metadata standards
+- Streamline metadata management workflows
+- Search among all entities in data.all based on attached metadata
+
+**Metadata Form lifecycle and usage**
+
```

Comment:
nit: define what visibility means. It is more or less obvious, but let's make it obvious for someone that is new to data.all

Reply:
I added a paragraph for that.